test/async_worker.cc

```diff
@@ -117,6 +119,49 @@ class EchoWorker : public AsyncWorker {
   std::string echo;
 };
 
+std::condition_variable cv;
```

Comment:
Might be worth moving these variables to be members of `FailCancelWorker` instead.

Reply:
Yep I agree. Made new changes